core/src/main/java/com/scalar/db/api/DistributedStorage.java

```diff
@@ -121,6 +121,16 @@ public interface DistributedStorage {
    */
   Scanner scan(Scan scan) throws ExecutionException;
 
+  /**
+   * Retrieves all the records from the storage with the specified {@link ScanAll} command and
+   * returns {@link Scanner} to iterate the results.
+   *
+   * @param scanAll a {@code ScanAll} command
+   * @return {@link Scanner} to iterate results
+   * @throws ExecutionException if the operation failed
+   */
+  Scanner scanAll(ScanAll scanAll) throws ExecutionException;
```

Comment:
I didn't think we need to add a new method for the `ScanAll` operations, and I thought we can use the existing `scan` method. What do you think?

Reply:
Ok, I misunderstood your design idea. I will remove this method.